generative_data_prep/data_prep/pipeline.py

```diff
@@ -384,6 +409,12 @@ def pipeline_main(  # noqa: C901
         Metrics associated with tokenization
     """
     # print input file information
+    dataset_metadata_json = {
```

Comment:
nit: this should be an object, either using the "DatasetMetadata"  or in corporate it into the existing metrics class? It is bad practice to create objects in python using dictionaries. Especially when the attributes are so well defined like here (you know the types and such)  

We can do this in a future PR, if you agree then just open an issue and then create a PR for it in the future.